### DISCUSSION TITLE
Help - Amplify + RDS MSSQL + Error=Callback#

### DISCUSSION THREAD

**Discussion by preludeofme** [nextauthjs/next-auth#2865] [Help]
I am trying to deploy my react/nextjs/nextauth application to AWS using Amplify and an RDS MS Sql Server. I am able to get everything working locally but when I deploy to Amplify and try to authenticate with the google provider, it just returns me to the sign in page with the error message in the url "Error=Callback#". I am not sure how to debug and get any specifics out of the error beyond that. 

I've got all my environment variables set up properly (validated they're coming through ok), i've set up a VNC, Security Group, and domain, CORS... everything i could think of based on various tutorials but i'm not sure why i can't get it working. I turned debugging on through the nextauth config but i'm not sure where to view the output of the debugging, there's nothing in the browser console and i'm not sure where to find the console for the amplify server (if there is any). i'm still a bit new to AWS and nextauth 

Here is my [...nextauth].tsx file: 

```js
import NextAuth from "next-auth";
import Providers from "next-auth/providers";

export default NextAuth({
  providers: [
    Providers.Email({
      name: "Email",
      server: {
        port: 465,
        host: "smtp.gmail.com",
        secure: true,
        auth: {
          user: process.env.EMAIL_SERVER_USER,
          pass: process.env.EMAIL_SERVER_PASSWORD,
        },
        tls: {
          rejectUnauthorized: false,
        },
      },
      from: process.env.EMAIL_FROM,
    }),
    Providers.Google({
      clientId: process.env.GOOGLE_ID,
      clientSecret: process.env.GOOGLE_SECRET,
    }),
  ],

  database: {
    type: "mssql",
    url: process.env.DATABASE_URL,
    port: 1433,
    database: "dbName",
    schema: "auth",
    extra: {
      validateConnection: false,
      trustServerCertificate: true,
    },
    authentication: {
      type: "default",
      options: { userName: process.env.DB_USER, password: process.env.DB_PW },
    },
  },
  secret: process.env.SECRET,

  session: {
    jwt: false,
    maxAge: 30 * 24 * 60 * 60, // 30 days

  },


  jwt: {
  },

  pages: {
  },

  callbacks: {
    session(session, user) {
      console.log(session, user);
      if (user.id != null) {
        console.log("callback", user.id, session);
        session.userId = user.id;
      }

      return session;
    },

  },


  // Enable debug messages in the console if you are having problems
  debug: true,
});
```

What other info can i post to help?

**Comment by balazsorban44** [comment]
could you please post the logs from your console?

When does the error occur? can you reproduce it locally, or only when deployed?

**Reply by preludeofme** [reply]
So there aren't really anything in my browser console, just the "Flip script injected!" and "Navigated to https://accounts.google.com/o/oauth2....." and back to my app.

It only happens when i deploy it, locally it works fine. That's what is frustrating.

**Reply by balazsorban44** [reply]
I was thinking of the server logs on your deployment. the error above suggests the problem is happening there.

**Reply by preludeofme** [reply]
unfortunately i'm a noob on aws and can't seem to find the server logs, do you know how to find them in amplify/aws?

**Reply by balazsorban44** [reply]
no, I don't use those services

**Reply by preludeofme** [reply]
ok i'll dig around and see what i can get back to you as far as logs, appreciate it

**Reply by preludeofme** [reply]
So I found where to find the server logs and I'm not seeing anything come through after I authenticate through google/email. Is there another way to troubleshoot that you can think of?

**Reply by balazsorban44** [reply]
I would make sure that all environment variables are set correctly. (`NEXTAUTH_URL`, your client id and secret, etc.) and that you configured the redirect URL correctly in Google.

**Reply by preludeofme** [reply]
Yeah I doublechecked all of them and it all looks like it's working properly. I'll maybe try turning off the mssql side of things and see if it works.
